### PR TITLE
[WIP] bpo-32654: Fixes for Android API 19

### PR DESCRIPTION
**Pull request by vstinner**
* Declare mmap() prototype on API < 21
* Avoid setlocale(LC_ALL, NULL) in Py_Main() on API < 20
* pystrtod.c: avoid localeconv() on API < 21
* _Py_GetLocaleconvNumeric(): avoid localeconv() on API < 20, use
  hardcoded values.

https://bugs.python.org/issue32654

### CODE REVIEW

**Comment by vstinner** [comment]
@yan12125: Would you mind to review and/or test this PR?

I'm not sure that we should support Android API 19, see: https://bugs.python.org/issue32654#msg310682

**Comment by yan12125** [comment]
> @yan12125: Would you mind to review and/or test this PR?

Sorry, I may not be able to for now. I don't have a working android-19 device, and I don't think testing on an emulator can give trustworthy results. In my experience there are lots of bugs occur on emulators only.

**Comment by vstinner** [comment]
> Sorry, I may not be able to for now. I don't have a working android-19 device, and I don't think testing on an emulator can give trustworthy results. In my experience there are lots of bugs occur on emulators only.

The question is not if this PR is enough to fix all API 19 issues, just if it doesn't make things worse :-D

**Comment by yan12125** [comment]
Oh, of course it won't be worse than crash immediately after startup :)

I can have a check on emulators, possibly this weekend.

File: Python/pystrtod.c [hunk 1]
```diff
@@ -177,8 +176,13 @@ _PyOS_ascii_strtod(const char *nptr, char **endptr)
 
     fail_pos = NULL;
 
-    locale_data = localeconv();
+#if defined(__ANDROID_API__) and __ANDROID_API__ < 21
```
Comment (yan12125) [inline Python/pystrtod.c]: and => &&

It's not Python :)

Comment (vstinner) [reply Python/pystrtod.c]: Oops :-) Fixed.

**Pull request closed without merging**